src/test/scala/ai/privado/languageEngine/java/passes/config/JavaYamlLinkerPassTest.scala

```diff
@@ -0,0 +1,199 @@
+package ai.privado.languageEngine.java.passes.config
+
+import ai.privado.cache.{RuleCache, TaggerCache}
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.languageEngine.java.tagger.sink.JavaAPITagger
+import ai.privado.languageEngine.java.tagger.source.IdentifierTagger
+import ai.privado.model.{
+  CatLevelOne,
+  ConfigAndRules,
+  Constants,
+  FilterProperty,
+  Language,
+  NodeType,
+  RuleInfo,
+  SystemConfig
+}
+import ai.privado.utility.PropertyParserPass
+import better.files.File
+import io.joern.dataflowengineoss.layers.dataflows.{OssDataFlow, OssDataFlowOptions}
+import io.joern.javasrc2cpg.Config
+import io.joern.javasrc2cpg.JavaSrc2Cpg
+import io.joern.x2cpg.X2Cpg.applyDefaultOverlays
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.semanticcpg.layers.LayerCreatorContext
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.matchers.should.Matchers.should
+import org.scalatest.wordspec.AnyWordSpec
+import io.shiftleft.semanticcpg.language.*
+import ai.privado.languageEngine.java.language.*
+
+class JavaYamlLinkerPassTest extends JavaYamlLinkerPassTestBase {
+  override val yamlFileContents =
+    """
+      |config:
+      |  default:
+      |    API_URL: http://exampleKubernetesService
+      |""".stripMargin
+
+  override val codeFileContents =
+    """
+      |import org.apache.http.HttpResponse;
+      |import org.apache.http.client.HttpClient;
+      |import org.apache.http.client.methods.HttpGet;
+      |import org.apache.http.impl.client.HttpClients;
+      |
+      |public class APICaller {
+      |  private String apiUrl;
+      |
+      |  public makeCall() {
+      |    apiUrl = System.getenv("API_URL");
```

Comment:
This is used to get Environment variables, are we saying this operation happens only with Yaml file in Java?